### PR TITLE
Expose the default json serializer

### PR DESCRIPTION
**Pull request by SerayaEryn**
I just had the situation where I needed to access the default json serializer. This PR exposes the `serialize` function at the reply.

### CODE REVIEW

**Comment by mcollina** [comment]
Can you please add a unit test showing how would you use `serialize`?

**Comment by SerayaEryn** [comment]
I want to write a serializer that uses `serialize` function internally and compresses the output of `serialize`. 
 In the `onSend` Hook the required data is available.

I'll add a unit test tomorrow.

**Comment by delvedor** [comment]
The problem is that `serialize` is an internal function, and it could change at any time.
I'm not sure that expose it is a good idea.

**Comment by SerayaEryn** [comment]
I added a simple unit test to show what I would do with `serialize`.

What about instead of exposing `serialize` expose a function that is calling it like this to avoid exposing it directly?

```js
Reply.prototype.serialize = function (reply, payload) {
  return serialize(reply.context, payload, reply.res.statusCode)
}
```

**Comment by delvedor** [comment]
@SerayaEryn I like that solution! :)

**Comment by SerayaEryn** [comment]
I updated the PR to not expose the `serialize` function directly.

File: docs/Reply.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ Reply is a core Fastify object that exposes the following functions:
 - `.header(name, value)` - Sets the headers.
 - `.type(value)` - Sets the header `Content-Type`.
 - `.redirect([code,] url)` - Redirect to the specified url, the status code is optional (default to `302`).
+- `.serialize(reply, payload)` - The default json serializer.
```
Comment (jsumners) [inline docs/Reply.md]: It's not entirely clear to me why this would need both parameters. The documentation should make it clear to me.

**Comment by SerayaEryn** [comment]
@jsumners I updated the docs. I hope it is better now.

**Comment by jsumners** [comment]
@SerayaEryn not really. Why does it need the reply if it is only serializing the payload?

**Comment by SerayaEryn** [comment]
@jsumners I removed the reply from the parameters of the function.

**Pull request merged**